src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -77,21 +104,95 @@ export const CollectionNfts = ({ contractAddress }: CollectionNftsProps) => {
 
   const [uniformHeight, setUniformHeight] = useState<UniformHeight>(UniformHeights.unset)
   const [currentTokenPlayingMedia, setCurrentTokenPlayingMedia] = useState<string | undefined>()
+  const [isFiltersExpanded, setFiltersExpanded] = useFiltersExpanded()
+  const oldStateRef = useRef<CollectionFilters | null>(null)
+  const isMobile = useIsMobile()
 
   const collectionNfts = useMemo(() => {
     if (!collectionAssets || !AssetsFetchSuccess) return undefined
 
     return collectionAssets.pages.flat()
   }, [collectionAssets, AssetsFetchSuccess])
 
+  const rarityStatusCache = new Map<string, boolean>()
+  const hasRarity = getRarityStatus(rarityStatusCache, collectionStats?.address, collectionNfts)
+
+  const sortDropDownOptions: DropDownOption[] = useMemo(
+    () =>
+      hasRarity
+        ? [
+            {
+              displayText: 'Low to High',
+              onClick: () => setSortBy(SortBy.LowToHigh),
+              icon: <NonRarityIcon width="28" height="28" color={vars.color.blue400} />,
+              reverseIndex: 2,
+            },
+            {
+              displayText: 'High to Low',
+              onClick: () => setSortBy(SortBy.HighToLow),
+              icon: <NonRarityIcon width="28" height="28" color={vars.color.blue400} />,
+              reverseIndex: 1,
+            },
+            {
+              displayText: 'Rare to Common',
+              onClick: () => setSortBy(SortBy.RareToCommon),
+              icon: <RarityIcon width="28" height="28" color={vars.color.blue400} />,
+              reverseIndex: 4,
+            },
+            {
+              displayText: 'Common to Rare',
+              onClick: () => setSortBy(SortBy.CommonToRare),
+              icon: <RarityIcon width="28" height="28" color={vars.color.blue400} />,
+              reverseIndex: 3,
+            },
+          ]
+        : [
+            {
+              displayText: 'Low to High',
+              onClick: () => setSortBy(SortBy.LowToHigh),
+              icon: <NonRarityIcon width="28" height="28" color={vars.color.blue400} />,
+              reverseIndex: 2,
+            },
+            {
+              displayText: 'High to Low',
+              onClick: () => setSortBy(SortBy.HighToLow),
+              icon: <NonRarityIcon width="28" height="28" color={vars.color.blue400} />,
+              reverseIndex: 1,
+            },
+          ],
+    [hasRarity, setSortBy]
+  )
+
   useEffect(() => {
     setUniformHeight(UniformHeights.unset)
   }, [contractAddress])
 
-  if (!collectionNfts) {
-    // TODO: collection unavailable page
-    return <div>No CollectionAssets</div>
-  }
+  useEffect(() => {
```

Comment:
We should still have the catch for the unavailable page, right now it's just a black page with the filter menu. You can test by inputting a bad collectionAddress in the url